Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs

```diff
@@ -0,0 +1,176 @@
+            {
+                ConsolePopup(args.Session, Loc.GetString("shipyard-console-invalid-vessel", ("vessel", args.Vessel)));
+                PlayDenySound(uid, component);
+                return;
+            }
+
+            if (vessel.Price <= 0)
+                return;
+
+            var station = _station.GetOwningStation(uid);
+            var bank = GetBankAccount(station);
+
+            if (bank == null)
+                return;
+
+            if (bank.Balance <= vessel.Price)
+            {
+                ConsolePopup(args.Session, Loc.GetString("cargo-console-insufficient-funds", ("cost", vessel.Price)));
+                PlayDenySound(uid, component);
+                return;
+            }
+
+            if (!TryPurchaseVessel(bank, vessel, out var shuttle) || shuttle == null)
+            {
+                PlayDenySound(uid, component);
+                return;
+            }
+
+            _cargo.DeductFunds(bank, vessel.Price);
+            var channel = _prototypeManager.Index<RadioChannelPrototype>("Command");
+            _radio.SendRadioMessage(uid, Loc.GetString("shipyard-console-docking", ("vessel", vessel.Name.ToString())), channel);
+            PlayConfirmSound(uid, component);
+
+            var newState = new ShipyardConsoleInterfaceState(
+                bank.Balance,
+                true);
+
+            _ui.TrySetUiState(uid, ShipyardConsoleUiKey.Shipyard, newState); 
+        }
+        
+        private void OnConsoleUIOpened(EntityUid uid, SharedShipyardConsoleComponent component, BoundUIOpenedEvent args)
+        {
+            if (!args.Session.AttachedEntity.HasValue)
+                return;
+
+            var station = _station.GetOwningStation(uid);
+            var bank = GetBankAccount(station);
+
+            if (bank == null)
+                return;
+
+            var newState = new ShipyardConsoleInterfaceState(
+                bank.Balance,
+                true);
+
+            _ui.TrySetUiState(uid, ShipyardConsoleUiKey.Shipyard, newState);
+        }
+
+        private void ConsolePopup(ICommonSession session, string text)
+        {
+            if (session.AttachedEntity is { Valid : true } player)
+                _popup.PopupEntity(text, player);
+        }
+
+        private void PlayDenySound(EntityUid uid, SharedShipyardConsoleComponent component)
+        {
+            _audio.PlayPvs(_audio.GetSound(component.ErrorSound), uid);
+        }
+        
+        private void PlayConfirmSound(EntityUid uid, SharedShipyardConsoleComponent component)
+        {
+            _audio.PlayPvs(_audio.GetSound(component.ConfirmSound), uid);
+        }
+
+        private bool TryPurchaseVessel(StationBankAccountComponent component, VesselPrototype vessel, out ShuttleComponent? deed)
+        {
+            var stationUid = _station.GetOwningStation(component.Owner);
+
+            if (component == null || vessel == null || vessel.ShuttlePath == null || stationUid == null)
```

Comment:
If component, vessel, or the path can be null here, you've been ignoring nullables. Redundant checks.